exist-core/src/main/java/org/exist/xquery/DynamicFunctionCall.java

```diff
@@ -96,4 +107,31 @@ public void resetState(final boolean postOptimization) {
         functionExpr.resetState(postOptimization);
         arguments.forEach(arg -> arg.resetState(postOptimization));
     }
+
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        if (functionExpr != null) {
+            builder.append(functionExpr);
+        } else {
+            builder.append("DynamicFunctionCall{arguments=");
+        }
+
+        builder.append('(');
+        if (arguments != null) {
+            for (int i = 0; i < arguments.size(); i++) {
+                if (i > 0) {
+                    builder.append(", ");
+                }
+                builder.append(arguments.get(i).toString());
+            }
```

Comment:
Following would reduce the amount of calls:
```
boolean firstArgument = true;
for (final Expression argument : arguments) {
    builder.append(argument.toString());
    if (firstArgument) {
        firstArgument = false;
    } else {
        builder.append(", ");
    }
}
```

Reply:
@reinhapa I think this will lead to one too many comma characters, i.e. a dangling comma at the end... or am I mistaken?

Reply:
@adamretter got my order wrong, should be:
```
boolean firstArgument = true;
for (final Expression argument : arguments) {
    if (firstArgument) {
        firstArgument = false;
    } else {
        builder.append(", ");
    }
    builder.append(argument.toString());
}
```

Reply:
So.... wassup?